src/main/java/com/blackducksoftware/integration/hub/alert/digest/DigestNotificationProcessor.java

```diff
@@ -51,74 +58,115 @@
-                categoryBuilderMap = projectBuilder.getCategoryBuilderMap();
-            }
-            // get the category data object to be able to add items.
-            CategoryDataBuilder categoryData;
-            final NotificationCategoryEnum categoryKey = NotificationCategoryEnum.valueOf(entry.getNotificationType());
-            if (!categoryBuilderMap.containsKey(categoryKey)) {
-                categoryData = new CategoryDataBuilder();
-                categoryData.setCategoryKey(categoryKey.name());
-                categoryBuilderMap.put(categoryKey, categoryData);
-            } else {
-                categoryData = categoryBuilderMap.get(categoryKey);
-            }
-            int count = 1;
-            final Map<String, Object> dataSet = new HashMap<>();
-            if (categoryKey == NotificationCategoryEnum.HIGH_VULNERABILITY || categoryKey == NotificationCategoryEnum.MEDIUM_VULNERABILITY || categoryKey == NotificationCategoryEnum.LOW_VULNERABILITY) {
-                count = entry.getVulnerabilityList().size();
-                dataSet.put(VulnerabilityCache.VULNERABILITY_ID_SET, getVulnerabilityIdSet(entry));
-            }
+    private Collection<UserNotificationWrapper> createCateoryDataMap(final DigestTypeEnum digestType, final Collection<NotificationEntity> notificationList) {
+        final Map<String, Map<String, ProjectDataBuilder>> userProjectMap = createUserProjectMap(digestType, notificationList);
+        final Collection<UserNotificationWrapper> dataList = new LinkedList<>();
+        userProjectMap.entrySet().forEach(userMapEntry -> {
+            final String username = userMapEntry.getKey();
+            try {
+                final HubUsersEntity userEntity = hubUsersRepository.findByUsername(username);
+                if (userEntity != null) {
+                    final Map<String, ProjectDataBuilder> projectDataMap = userMapEntry.getValue();
+                    final Set<ProjectData> projectDataSet = new LinkedHashSet<>();
+                    projectDataMap.values().forEach(projectDataBuilder -> {
+                        projectDataSet.add(projectDataBuilder.build());
+                    });
 
-            if (StringUtils.isNotBlank(entry.getPolicyRuleName())) {
-                dataSet.put(ItemTypeEnum.RULE.name(), entry.getPolicyRuleName());
+                    dataList.add(new UserNotificationWrapper(userEntity.getId(), projectDataSet));
+                }
+            } catch (final NoResultException ex) {
+                logger.debug("user {} could not be found in the configuration", username);
+                logger.debug("Cause:", ex);
             }
+        });
+        return dataList;
+    }
+
+    private Map<String, Map<String, ProjectDataBuilder>> createUserProjectMap(final DigestTypeEnum digestType, final Collection<NotificationEntity> notificationList) {
+        final Map<String, Map<String, ProjectDataBuilder>> userProjectMap = new LinkedHashMap<>();
```

Comment:
another map in map where we could probably cut out some code through an extra class